solutions/CSES Subarray Sums II.mdx

```diff
@@ -0,0 +1,179 @@
+---
+id: cses-1661
+source: CSES
+title: Subarray Sums II
+author: Qi Wang
+---
+
+# Problem
+
+We are asked to find the number of subarrays that sum up to $\texttt{x}$ given the size of the array and its elemets.
+
+<Spoiler title="Solution">
+
+We can create a prefix sum and check whether it is equal to $\texttt{x}$. If it is, then increment the $\texttt{res}$. Furthermore, we can have a map that keeps track of these prefix sums and when the prefix sums exceeds $\texttt{x}$, we can check whether there are subarrays of $\texttt{x} - \texttt{prefix sum}$. This will ensure that we can remove that amount of the subarray that has $\texttt{x} - \texttt{prefix sum}$ from our current subarray to build our answer.
```

Comment:
Define what $\texttt{res}$ is and then remove "the" from 

> If it is, then increment the $\texttt{res}$.

Reply:
> we can check whether there are subarrays of $\texttt{x} - \texttt{prefix sum}$

subarrays with prefix sum equal to $\texttt{x} - \texttt{prefix sum}$. Also,  $\texttt{prefix sum}$ doesn't necessarily exceed $\texttt{x}$.

Reply:
> This will ensure that we can remove that amount of the subarray that has $\texttt{x} - \texttt{prefix sum}$ from our current subarray to build our answer.

You mean - we can remove one prefix from our current prefix?